test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java

```diff
@@ -133,7 +135,7 @@ private PathTestHelper() {
      */
     public static Path getTestClassesLocation(Class<?> testClass) {
         String classFileName = testClass.getName().replace('.', File.separatorChar) + ".class";
-        URL resource = testClass.getClassLoader().getResource(testClass.getName().replace('.', '/') + ".class");
+        URL resource = testClass.getClassLoader().getResource(fromClassNameToResourceName(testClass.getName()));
```

Comment:
Maybe there should also be a variant that accepts `Class<?>`? Then we can be sure to reject hidden classes, which will surely give bizarre results if passed through this method.